Assignment/Assignment.Tests/SampleDataTests.cs

```diff
@@ -83,11 +86,74 @@ public void GetUniqueSortedListOfStatesGivenCsvRows_ValidParameters_ChecksIfColl
     public void GetAggregateSortedListOfStatesUsingCsvRows_WithCSV_ReturnsCommaSeperatedStatesList()
     {
         //Arrange
-        SampleData initClass = new SampleData();
+        SampleData initClass = new();
         string expectedString = "AL,AZ,CA,DC,FL,GA,IN,KS,LA,MD,MN,MO,MT,NC,NE,NH,NV,NY,OR,PA,SC,TN,TX,UT,VA,WA,WV";
         string actualString = initClass.GetAggregateSortedListOfStatesUsingCsvRows();
         //Assert
         Assert.AreEqual(expectedString.Length, actualString.Length);
         Assert.AreEqual(expectedString, actualString);
     }
+
+    [TestMethod]
+    public void People_Ordered_Matches()
+    {
+        //Arrange
+        SampleData initClass = new();
+
+        //Act
+        IEnumerable<IPerson> people = initClass.People;
+
+        //Assert
+        Assert.AreEqual(50, people.Count());
+
+        Assert.AreEqual("Arthur Myles", $"{people.First().FirstName} {people.First().LastName}");
+        Assert.AreEqual("Amelia Toal", $"{people.Last().FirstName} {people.Last().LastName}");
+    }
+
+    [TestMethod]
+    public void FilterByEmailAddress_GivenExactEmail_MatchesName()
+    {
+        //Arrange
+        SampleData initClass = new();
+        Predicate<string> filter = (string email) => email == "pjenyns0@state.gov";
+
+        //Act
+        IEnumerable<(string FirstName, string LastName)> filteredData = initClass.FilterByEmailAddress(filter);
+
+
+        //Assert
+        Assert.AreEqual(1, filteredData.Count());
+        Assert.AreEqual<(string FirstName, string LastName)>(("Priscilla", "Jenyns"), filteredData.First());
+    }
+
+    [TestMethod]
+    public void FilterByEmailAddress_GivenContains_ReturnsManyMatching()
+    {
+        //Arrange
+        SampleData initClass = new();
+        Predicate<string> filter = (string email) => email.Contains(".gov");
+
+        //Act
+        IEnumerable<(string FirstName, string LastName)> filteredData = initClass.FilterByEmailAddress(filter);
+
+
+        //Assert
+        Assert.AreEqual(5, filteredData.Count());
+        Assert.AreEqual<(string FirstName, string LastName)>(("Arthur", "Myles"), filteredData.First());
+        Assert.AreEqual<(string FirstName, string LastName)>(("Amelia", "Toal"), filteredData.Last());
+    }
+
+    [TestMethod]
+    public void GetAggregateListOfStatesGivenPeopleCollection_GivenPeople_ReturnsListOfStates()
```

Comment:
Requirement 6 last bullet wants for testing this method to at a minimum use the ISampleData.GetUniqueSortedListOfStatesGivenCsvRows to validate your result.